### ISSUE TITLE
Bug: Linker errors when building latest version from GIT

### ISSUE THREAD

**Issue by Elmi77** [assimp/assimp#4752]
I'm trying to build assimp (latest version from GIT) using Visual Studio 2019 and the provided assimp.sln but get loads of linker errors:

```
LNK2001 unresolved external symbol "public: __cdecl aiVector3t<float>::aiVector3t<float>(class aiVector3t<float> const &)" (??0?$aiVector3t@M@@QEAA@AEBV0@@Z)   assimp  F:\wxWidgets-3\demos\assimp\code\Importer.obj   1   
LNK2001 unresolved external symbol "public: __cdecl aiColor4t<float>::aiColor4t<float>(class aiColor4t<float> const &)" (??0?$aiColor4t@M@@QEAA@AEBV0@@Z)   assimp  F:\wxWidgets-3\demos\assimp\code\X3DImporter_Rendering.obj  1   
LNK2001 unresolved external symbol "public: __cdecl aiVector2t<float>::aiVector2t<float>(class aiVector2t<float> const &)" (??0?$aiVector2t@M@@QEAA@AEBV0@@Z)   assimp  F:\wxWidgets-3\demos\assimp\code\XmlSerializer.obj  1   
LNK2019 unresolved external symbol "public: __cdecl aiVector3t<float>::aiVector3t<float>(class aiVector3t<float> const &)" (??0?$aiVector3t@M@@QEAA@AEBV0@@Z) referenced in function "public: struct aiFace * __cdecl std::vector<struct aiFace,class std::allocator<struct aiFace> >::_Emplace_reallocate<struct aiFace const &>(struct aiFace * const,struct aiFace const &)" (??$_Emplace_reallocate@AEBUaiFace@@@?$vector@UaiFace@@V?$allocator@UaiFace@@@std@@@std@@QEAAPEAUaiFace@@QEAU2@AEBU2@@Z)    assimp  F:\wxWidgets-3\demos\assimp\code\X3DGeoHelper.obj   1   
LNK2019 unresolved external symbol "public: __cdecl aiVector2t<double>::aiVector2t<double>(class aiVector2t<double> const &)" (??0?$aiVector2t@N@@QEAA@AEBV0@@Z) referenced in function "private: void __cdecl std::deque<class aiVector2t<double>,class std::allocator<class aiVector2t<double> > >::_Emplace_back_internal<class aiVector2t<double> const &>(class aiVector2t<double> const &)" (??$_Emplace_back_internal@AEBV?$aiVector2t@N@@@?$deque@V?$aiVector2t@N@@V?$allocator@V?$aiVector2t@N@@@std@@@std@@AEAAXAEBV?$aiVector2t@N@@@Z)   assimp  F:\wxWidgets-3\demos\assimp\code\IFCOpenings.obj    1   

VS builds in C++17 mode and with the preprocessor options

 %(PreprocessorDefinitions);WIN32;_WINDOWS;NDEBUG;ASSIMP_BUILD_DLL_EXPORT;ASSIMP_BUILD_NO_M3D_IMPORTER;ASSIMP_BUILD_NO_M3D_EXPORTER;WIN32_LEAN_AND_MEAN;UNICODE;_UNICODE;ASSIMP_BUILD_NO_EXPORT;ASSIMP_BUILD_NO_C4D_IMPORTER;ASSIMP_IMPORTER_GLTF_USE_OPEN3DGC=1;RAPIDJSON_HAS_STDSTRING=1;RAPIDJSON_NOMEMBERITERATORCLASS;_SCL_SECURE_NO_WARNINGS;_CRT_SECURE_NO_WARNINGS;OPENDDLPARSER_BUILD;CMAKE_INTDIR="Release";assimp_EXPORTS

```
The related headers/classes of the objects missing during linking are there and are built (at least as I can see, I'm not sure regarding all this template stuff used there).

So...any idea what could be wrong and how to fix this linker issue?

**Comment by kimkulling** [comment]
Is the sld-file the generated one coming from a cmake run?

**Comment by Elmi77** [comment]
Yes, the SLN and the VCXPRJ come from an CMake run - and they used to work. It seems one of the last GIT-updates caused the problems

**Comment by Elmi77** [comment]
Update: even recreating the project files with CMake and rebuilding the whole stuff does not solve the issue. May be it has something to do with my modifications in CMakeLists.txt? This is what I have set according to the manual:

OPTION( ASSIMP_DOUBLE_PRECISION
  "Set to ON to enable double precision processing"
  ON
)

OPTION( ASSIMP_NO_EXPORT
  "Disable Assimp's export functionality."
  ON
)

OPTION ( ASSIMP_BUILD_TESTS
  "If the test suite for Assimp is built in addition to the library."
  OFF
)

OPTION ( ASSIMP_WARNINGS_AS_ERRORS
  "Treat all warnings as errors."
  OFF
)

OPTION( ASSIMP_INJECT_DEBUG_POSTFIX
  "Inject debug postfix in .a/.so/.dll lib names"
  OFF
)

**Comment by kimkulling** [comment]
Could you please provide your used cmake command from the command line? And are you able to see the libs generated by the build in the in\debug folder or in the lib\debug folder?

**Comment by Elmi77** [comment]
Update: I tried compiling it with a fresh and clean checkout from GIT, with the default CMakeLists.txt and with the default settings that came with the VS project files. The result was exactly the same. So it seems, the current version in GIT can't - for whatever reason - be built with VisualStudio 2019

**Comment by Elmi77** [comment]
Update 2: when swithcing to the release tagged with "5.2.0" in GIT, the library buildsp roperly and the linker errors are gone.

**Comment by Elmi77** [comment]
Update 3: it seems "v5.2.2" is the last version that compiles well with Visual Studio 2019. "v5.2.3" fais with some parameter type errors in aiMaterial (so not the linker error this bugreport is about, but also some basic build problems)

**Comment by Elmi77** [comment]
After updating to the latest compiler version of VS2019, the issue seems to be solved, version 5.2.5 compiles without any problems.

**Comment by kimkulling** [comment]
Thanks for the update, I will close this issue.
